### PR TITLE
[vcpkg formatting] Turn off DeriveLineEnding

### PR DESCRIPTION
**Pull request by strega-nil**
This PR removes CRLFs from existing C++ files, and standardizes upon the line ending `\n`.

This is not necessarily a thing we wish to do, so I'm labelling it requires:discussion.

### CODE REVIEW

**Comment by cenit** [comment]
Adding to the discussion, what about doing the same thing (and enforcing the rule with .gitattributes) also for port files, control files and patches?
Also file permissions (some port files have executable bit on...)

**Comment by ras0219-msft** [comment]
We do like standardization, but we also don't want to waste people's time; an important factor that significantly reduces costs of imposing LF on code is that clang-format will automatically correct the issue without users even noticing (which we already require contributors to run anyway).

I think that before we move to enforce such conventions on port files, we should seek to have tooling that will automatically fix i t.

**Review by ras0219-msft — APPROVED** [review]

**Comment by cenit** [comment]
> I think that before we move to enforce such conventions on port files, we should seek to have tooling that will automatically fix it.

isn't it possible to do it using git & enforcing gitattributes file?

File: toolsrc/include/vcpkg/binarycaching.private.h [hunk 1]
```diff
@@ -1,60 +1,60 @@
...
+        std::string buf;
+
+    private:
+        XmlSerializer& emit_pending_indent();
+
+        int m_indent = 0;
+        bool m_pending_indent = false;
+    };
+
 }
```
Comment (strega-nil) [inline toolsrc/include/vcpkg/binarycaching.private.h:60]:
```suggestion
}

```

File: toolsrc/src/vcpkg-test/mockcmakevarsprovider.cpp [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
+    }
+
+    Optional<const std::unordered_map<std::string, std::string>&> MockCMakeVarProvider::get_tag_vars(
+        const PackageSpec& spec) const
+    {
+        auto it = tag_vars.find(spec);
+        if (it == tag_vars.end()) return nullopt;
+        return it->second;
+    }
 }
```
Comment (strega-nil) [inline toolsrc/src/vcpkg-test/mockcmakevarsprovider.cpp:28]:
```suggestion
}

```

**Pull request merged**